### DISCUSSION TITLE
http://localhost:3000 is not an accepted origin. (further occurrences of this error will be logged with level INFO)

### DISCUSSION THREAD

**Discussion by ManojVysyaraju** [miguelgrinberg/python-socketio#855] [Q&A] [answered]
Hi @miguelgrinberg 

After hosing, I am not able to initiate a connection to the WebSocket app from a remote application(react). I am able to connect using client.py but not from JS.

js script --> https://cdn.socket.io/4.4.1/socket.io.min.js

const sio = io('http://qsocket.somedomain.com/', { transports: ["websocket"] });
sio.on('connect', () => {
      console.log('connected');
      console.log('Trying to join room', id);
      sio.emit('join_room', id, (data) => {
        console.log(data);
      });
    });
    sio.on('connect_error', (msg) => {
      console.log(msg);
    });
Error from JS Side: websocket.js:54 WebSocket connection to 'ws://qsocket.somedomain.com/socket.io/?EIO=4&transport=websocket' failed

Error Log from app.py : http://localhost:3000 is not an accepted origin. (further occurrences of this error will be logged with level INFO)

App Version Info:
python-socketio = {extras = ["client"], version = "*"}
gunicorn = "*"
eventlet = "==0.30.2"
flask = "*"
injector = "*"
flask-injector = "*"
flask-cors = "*"

[app.py.txt](https://github.com/miguelgrinberg/python-socketio/files/7894183/app.py.txt)

**Comment by miguelgrinberg** [comment, marked as answer]
See the documentation regarding cross-origin (CORS) configuration: https://python-socketio.readthedocs.io/en/latest/api.html#server-class. The option is called `cors_allowed_origins` and in your case should be set to `http://localhost:3000` (as it appears in your error message.

**Reply by ManojVysyaraju** [reply]
sio = socketio.Server(cors_allowed_origins="*")
this worked. 
I thought it only belongs to The Engine.IO configuration but it worked for socketio.Server too.

Thank you

**Reply by miguelgrinberg** [reply]
You ignored my suggestions solution which is better and more secure:

    sio = socketio.Server(cors_allowed_origins="http://localhost:3000")
